src/silx/gui/plot/PlotToolButtons.py

```diff
@@ -624,13 +626,33 @@ class RulerToolButton(PlotToolButton):
     class RulerROI(LineROI):
         def __init__(self, parent=None):
             super().__init__(parent)
-            self._formatFunction = None
-
-        def registerFormatFunction(self, fct):
-            """fct is expected to be a function taking (startPoint, endPoint) as parameter"""
+            self._formatFunction: typing.Optional[
+                typing.Callable[
+                    [numpy.ndarray, numpy.ndarray], str
+                ]
+            ] = None
+            self.setColor("#001122")  # Only there to trig updateStyle
+
+        def registerFormatFunction(
```

Comment:
Would better be named `set` since it is a setter:
```suggestion
        def setFormatFunction(
```
but fine since this can be considered as private API